### PR TITLE
Bugfix/float builds

### PR DESCRIPTION
**Pull request by vlazzarini**
A number of issues were detected in the 32-bit floats builds. This PR deals with these:

- variable defined outside the USE_DOUBLE guards unused in floats build.
- table sizes need to be limited to 2^24 size in floats build because there is not enough range to properly handle larger numbers in MYFLT. Since data in score and orchestra is represented using MYFLT, it is not possible to correctly handle integers larger than 2^24. Table sizes cannot be accurately set beyond 2^24.
- Some tests were failing because double constants were assumed.
- lrint was not being searched for correctly and it was not being found on MacOs.

I also introduced `var:i floatsize` to return the size of MYFLT so code can be written to depend on this.
CSD test for max table size was edit to take this into account. A new unit test was added to make sure
large table sizes can be set properly in doubles version.

Finally, I have set alternate MacOS and Linux builds to use 32-bit floats instead of doubles so we can catch any issues that may arise in the future.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -597,10 +597,15 @@ if (NOT MATH_LIBRARY)
     set(MATH_LIBRARY "")
 endif()
 
+if(APPLE)
+include(CheckSymbolExists)
+check_symbol_exists(lrint math.h FOUND_LRINT)
```
Comment (kunstmusik) [inline CMakeLists.txt:602]: Any reason to change from HAVE_LRINT to FOUND_LRINT? All of our other checks use "HAVE_" as far as I understand.

File: tests/commandline/test.py [hunk 2]
```diff
@@ -177,7 +177,6 @@ def runTest():
     ["test_plusname.csd", "test +Name for instr name"],
     ["testnewline.csd", "test newline in statements"],
     ["testmidichannels.csd", "test use of mapped multiport channels"],
-    ["test_max_table_len.csd", "test max table length"],
```
Comment (kunstmusik) [inline tests/commandline/test.py:180]: Looks like this test was removed from the suite, was this intentional?

File: Engine/entry1.c [hunk 3]
```diff
@@ -166,6 +166,7 @@ OENTRY opcodlst_1[] = {
   { "init.a", S(ASSIGNM),0,       "mmmmmmmmmmmmmmmmmmmmmmmm", "m", mainit },
   { "nstrnum",   S(IREF_NUM),0,    "i",    ":InstrDef;",   (SUBR) get_instr_num },
   { "init.instr", S(ASSIGN) ,0,  ":InstrDef;", ":InstrDef;", (SUBR) copyVarGenericInit},
+  { "floatsize", S(ASSIGN) ,0, "i", "", myflt_size },
```
Comment (kunstmusik) [inline Engine/entry1.c:169]: Just a note, my first reaction to this was why not a variable? But I think I'm 50/50 on it as an opcode and seems fine.

File: CMakeLists.txt (hunk 1 above)
Comment (vlazzarini) [reply CMakeLists.txt:602]: It was the only I could make it work.

File: tests/commandline/test.py (hunk 2 above)
Comment (vlazzarini) [reply tests/commandline/test.py:180]: yes, I moved it to a unit test as I could make it test the difference between float build and double build better.

File: Engine/entry1.c (hunk 3 above)
Comment (vlazzarini) [reply Engine/entry1.c:169]: I thought opcode was easier to implement.

File: tests/commandline/test.py (hunk 2 above)
Comment (kunstmusik) [reply tests/commandline/test.py:180]: Probably should delete the unused test csd then

Comment (kunstmusik) [reply tests/commandline/test.py:180]: Maybe the current test_max_table_len.csd can be moved to the manual as an example to use for floatsize?

**Review by kunstmusik — APPROVED** [review]
Looks good, merging now, thanks!

**Comment by vlazzarini** [comment]
thanks, it will be good to keep an eye on 32bit, we  often forget about it.

**Pull request merged**